src/liboslexec/optexture.cpp

```diff
@@ -18,169 +18,199 @@
 
 #include "oslexec_pvt.h"
 #include <OSL/dual.h>
+#include <OSL/rs_free_function.h>
 
 
 OSL_NAMESPACE_ENTER
 namespace pvt {
 
 
-// Utility: retrieve a pointer to the ShadingContext's texture options
-// struct, also re-initialize its contents.
-OSL_SHADEOP void*
-osl_get_texture_options(void* sg_)
-{
-    ShaderGlobals* sg = (ShaderGlobals*)sg_;
-    TextureOpt* opt   = sg->context->texture_options_ptr();
-    new (opt) TextureOpt;
-    return opt;
+OSL_SHADEOP OSL_HOSTDEVICE void
+osl_init_texture_options(OpaqueExecContextPtr oec, void* opt)
+{
+    // TODO: Simplify when TextureOpt() has __device__ marker.
+    // new (opt) TextureOpt;
+    TextureOpt* o          = reinterpret_cast<TextureOpt*>(opt);
+    o->firstchannel        = 0;
+    o->subimage            = 0;
+    o->swrap               = TextureOpt::WrapDefault;
+    o->twrap               = TextureOpt::WrapDefault;
+    o->mipmode             = TextureOpt::MipModeDefault;
+    o->interpmode          = TextureOpt::InterpSmartBicubic;
+    o->anisotropic         = 32;
+    o->conservative_filter = true;
+    o->sblur               = 0.0f;
+    o->tblur               = 0.0f;
+    o->swidth              = 1.0f;
+    o->twidth              = 1.0f;
+    o->fill                = 0.0f;
+    o->missingcolor        = nullptr;
+    o->time                = 0.0f;
+    o->rnd                 = -1.0f;
+    o->samples             = 1;
+    o->rwrap               = TextureOpt::WrapDefault;
+    o->rblur               = 0.0f;
+    o->rwidth              = 1.0f;
+#ifdef OIIO_TEXTURESYSTEM_SUPPORTS_COLORSPACE
+    o->colortransformid = 0;
```

Comment:
Add comment that envlayout is member right after ....  // Simulate memberindex(colortransformid) + 1